src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs

```diff
@@ -5156,6 +5156,325 @@ .maxstack  3
 ");
         }
 
+        [Fact]
+        public void LocalFunctionAttribute()
+        {
+            var source = @"
+class A : System.Attribute { }
+
+class C
+{
+    public void M()
+    {
+        [A]
+        void local1()
+        {
+        }
+
+        [return: A]
+        void local2()
+        {
+        }
+
+        void local3([A] int i)
+        {
+        }
+
+        void local4<[A] T>()
+        {
+        }
+    }
+}
+";
+            CompileAndVerify(
+                source,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All),
+                parseOptions: TestOptions.RegularPreview,
+                symbolValidator: validate);
+
+            void validate(ModuleSymbol module)
```

Comment:
>void [](start = 12, length = 4)

Minor point: Can each of these `validate` methods be `static`?

Reply:
GetAttributeNames is an instance method so it's a little more of a hassle

Reply:
It looks like `GetAttributeNames` and `GetAttributeStrings` can be `static`.

---
In reply to: [343380814](https://github.com/dotnet/roslyn/pull/39226#discussion_r343380814) [](ancestors = 343380814)